### PR TITLE
review

### PR DESCRIPTION
**Pull request by DanGhanbari**

### CODE REVIEW

**Comment by DibaKarimi** [comment]
Hi Daniel,
Welldone !  Your project looks great.
I've check your page and have few ideas:
1- you can sort shows list section like the DropDown.
2- I chose "American masters" which it's episodes have not images but display image. I could not find out the source of the images
3- When I filter,  shows  "Displaying 3 / 300 episodes" is display , I think it needs to change to "Displaying 3 / 300 shows".

**Review by kmh256 — COMMENTED** [review]
Fantastic work! Well done 👍 I like the hover effect on the images and the layout of the episodes. The only things I noticed were that the number of episodes displayed is not updating when you select the show and there are some shows such as Wire in the Blood which do show results. Your code is very tidy and clear and I have learned a lot from reviewing it.

File: script.js [hunk 1]
```diff
@@ -1,12 +1,235 @@
...
+  }
+}
+
+// display All show on main page
+function displayShow() {
+  let showObj = getAllShows();
+  selectEpisodeElement.style.display = "none";
+  selectEpisodesLabel.style.display = "none";
+  for (let i = 0; i < showObj.length; i++) {
+    truncate;
```
Comment (Buroni) [inline script.js:165]: I think you intended to call the `truncate` function, i.e. `truncate()`.

File: script.js [hunk 2]
```diff
@@ -1,12 +1,235 @@
...
+      mainElement.appendChild(showDiv);
+      rootElem.appendChild(mainElement);
+
+      //Adding event listener for selected show
+      showDiv.addEventListener("click", () => {
+        reset();
+        fetch(`https://api.tvmaze.com/shows/${showObj[i].id}/episodes`)
+          .then((response) => {
+            if (!response.ok) {
+              throw new Error("something wrong");
```
Comment (Buroni) [inline script.js:195]: It's great that you added error handling, but can you think of a way to make the message more informative? There might be some more details in the `response` object about what went wrong.

File: script.js [hunk 3]
```diff
@@ -1,12 +1,235 @@
...
+            }
+            return response.json();
+          })
+          .then((data) => {
+            const allEpisodes = data;
```
Comment (Buroni) [inline script.js:200]: Why don't you just rename `data` to `allEpisodes`? I.e.:

```js
.then((allEpsidoes) => {
 // ...
}
```

File: script.js [hunk 4]
```diff
@@ -1,12 +1,235 @@
...
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+  reset();
+  selectShow();
+  displayShow();
+  searchBox("shows");
+}
+
+function addZero(x) {
+  if (x < 10) {
```
Comment (Buroni) [inline script.js:20]: Can you think of a way to shorten this to a [ternary](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator)?

File: script.js [hunk 5]
```diff
@@ -1,12 +1,235 @@
...
+    let showId = selectShowElement.value;
+    selectEpisodeElement.innerHTML = "";
+    fetch(`https://api.tvmaze.com/shows/${showId}/episodes`)
+      .then((response) => {
+        if (!response.ok) {
+          throw new Error("something wrong");
+        }
+        return response.json();
+      })
+      .then((data) => {
```
Comment (Buroni) [inline script.js:149]: Nice promise chaining

File: script.js [hunk 6]
```diff
@@ -1,12 +1,235 @@
...
+    });
+  }
+}
+
+//select show
+let selectShow = function () {
+  mainElement.innerHTML = "";
+  const showList = getAllShows();
+  showList.sort((a, b) => (a.name > b.name ? 1 : -1));
+  var showArr = [];
```
Comment (Buroni) [inline script.js:115]: Can you think of a way to do this with an array of objects rather than two separate arrays? I.e. a variable that looks like

```js
[
    {show: "show 1", id: 0 },
    {show: "another show", id: 1},
    // ...
];
```

File: style.css [hunk 7]
```diff
@@ -1,3 +1,121 @@
 #root {
-  color: red;
+  background-color: rgb(209, 206, 180);
+}
+
+/* search section on header */
+header {
```
Comment (Buroni) [inline style.css:6]: Good use of flex

File: script.js [hunk 8]
```diff
@@ -1,12 +1,235 @@
...
+    let showDiv = document.createElement("div");
+    showDiv.className = "showSection";
+    if (showObj[i].image !== null) {
+      showDiv.innerHTML = `<div class=" show d-flex flex-column align-items-center mt-4">
+         <div class=" d-flex flex-column align-items-center justify-content-center">
+            <h4 class="pb-2" role="button">${showObj[i].name}</h4>
+            <img class="w-100 bd-highlight img-thumbnail" src =${showObj[i].image.medium}>
+        </div>
+        <div class="d-flex justify-content-between">
+            <div class= "summary p-4">Summary: ${showObj[i].summary}</div>
```
Comment (Buroni) [inline script.js:176]: You've accessed `showObj[i]` 8 times here. In a situation where the array is very large, it will be quite expensive to look up the same array item 8 times and so will slow your application down. It's faster to assign it to an array, i.e. `const show = showObj[i];` then use `show` as much as you need to.

**Review by Buroni — COMMENTED** [review]
Nice job Daniel! The code looks good and the website works well. You've modularised your code really well into functions that have one purpose. For example `dropDown` to build the dropdown element, `makePageForEpisodes`, etc..  You've also showed a good understanding of promise chaining and DOM manipulation.

See my comments below for more specific feedback and notes for improvement.
